src/yaml/from_yaml.cr

```diff
@@ -21,16 +21,23 @@ def Array.from_yaml(string_or_io)
 end
 
 def Nil.new(pull : YAML::PullParser)
-  value = pull.read_scalar
-  if value.empty?
+  case value = pull.read_scalar
+  when "", "NULL", "Null", "null"
```

Comment:
This should not be necessary because libyaml recognizes null values and it looks wrong, because it would also recognize a quoted string `"null"` as nil value.